### PR TITLE
Added support for encrypted passwords

### PR DESCRIPTION
**Pull request by hrmohr**
The plugin will get credentials from the settings.xml file but it does not currently support encrypted passwords.

### CODE REVIEW

**Comment by davidxia** [comment]
@hrmohr Maybe update the README to explain how to use this?

**Comment by davidxia** [comment]
I was wondering more about the SecDispatcher. I couldn't find any docs on
this.

On Thu, May 7, 2015 at 11:16 AM, Mads Mohr Christensen <
notifications@github.com> wrote:

> The documentation can be found in the official Maven documentation
> <https://maven.apache.org/guides/mini/guide-encryption.html>
>
> Should I just add a link to this in the readme?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/docker-maven-plugin/pull/87#issuecomment-99905748>
> .
>



-- 
David Xia
Software Engineer
dxia@spotify.com

**Comment by hrmohr** [comment]
@davidxia If you are thinking about the role-hint then this is explained in https://jira.codehaus.org/browse/MNG-4384

File: src/main/java/com/spotify/docker/AbstractDockerMojo.java [hunk 1]
```diff
@@ -53,6 +55,9 @@
   @Component
   private Settings settings;
 
+  @Component(role = SecDispatcher.class, hint = "mng-4384")
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/AbstractDockerMojo.java]: What does this hint do?

File: src/test/java/com/spotify/docker/SettingsStub.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+public class SettingsStub extends Settings {
+
+    public SettingsStub() {
+        super();
+        Server server = new Server();
+        server.setId("docker-hub");
+        server.setUsername("dxia3");
+        // encrypted value is: SxpxdUQA2mvX7oj
```
Comment (davidxia) [inline src/test/java/com/spotify/docker/SettingsStub.java]: Shouldn't this comment be "// plaintext value is: SxpxdUQA2mvX7oj"?

File: src/main/java/com/spotify/docker/AbstractDockerMojo.java (hunk 1 above)
Comment (davidxia) [inline src/main/java/com/spotify/docker/AbstractDockerMojo.java]: Oh nvm I see your comment above.

**Comment by davidxia** [comment]
@hrmohr sorry for the long delay. I'm not that familiar with SecDispatcher and if they have docs, I didn't find them.

Everything looks good. This change won't require users to encrypt their passwords right? They simply don't create a `~/.m2/settings-security.xml`?

**Comment by hrmohr** [comment]
@davidxia No problem. I have added some basic documentation just as you asked.

Maven only considers passwords enclosed in curly braces as encrypted. So encryption is not required but controlled by the format.

**Comment by hrmohr** [comment]
Eh, why did the build fail because of a long javadoc line?

**Comment by davidxia** [comment]
Hey thanks for adding to the docs. We have a check style plugin that fails
if there are lines of more than 100 chars.

On Saturday, August 8, 2015, Mads Mohr Christensen <notifications@github.com>
wrote:

> Eh, why did the build fail because of a long javadoc line?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spotify/docker-maven-plugin/pull/87#issuecomment-128970182>
> .
>


-- 
David Xia
Software Engineer
dxia@spotify.com

**Comment by davidxia** [comment]
@hrmohr Thanks!

**Pull request merged**